### ISSUE TITLE
[FEAT]: Notifications: Update `hmi-server` to interact with Job Schedulers

### ISSUE THREAD

**Issue by YohannParis** [DARPA-ASKEM/terarium#2069]
### Description
We need to enhance our `hmi-server` to interact directly with all the different job schedulers (`knowledge-middleware`, and simulation services) in order to create and update deferred tasks. This will allow us to efficiently manage and schedule tasks for execution. The integration should be modular, allowing us to support multiple job schedulers in the future.

### Tasks
- [ ] Implement a modular integration service that supports multiple job schedulers.
- [ ] Implement logic to create and update deferred tasks using the selected job scheduler(s).
- [ ] Ensure that the server can monitor the status of scheduled tasks and update the database accordingly.
- [ ] Implement error handling and retries for scheduled tasks.
- [ ] Create documentation for configuring and using the job scheduler integration.

### Acceptance Criteria
- The server is able to interact with the selected job scheduler(s) for creating, updating, and monitoring deferred tasks.
- The integration is modular and can be extended to support additional job schedulers in the future.
- Task scheduling and execution are reliable, with error handling and retries in place.
- Documentation is available for configuring and using the job scheduler integration.

### Additional Information
- Related Issues: depend on #2068
